src/openzeppelin/access/ownable.cairo

```diff
@@ -0,0 +1,54 @@
+#[contract]
+mod Ownable {
+    use starknet::ContractAddress;
+    use starknet::ContractAddressZeroable;
+    use starknet::contract_address_const;
+    use starknet::get_caller_address;
+    use zeroable::Zeroable;
+
+    struct Storage {
+        _owner: ContractAddress
+    }
+
+    #[event]
+    fn OwnershipTransferred(previous_owner: ContractAddress, new_owner: ContractAddress) {}
+
+    #[internal]
+    fn initializer() {
+        let caller: ContractAddress = get_caller_address();
+        _owner::write(caller);
+    }
+
+    #[internal]
+    fn assert_only_owner() {
+        let owner: ContractAddress = _owner::read();
+        let caller: ContractAddress = get_caller_address();
+        assert(!caller.is_zero(), 'Caller is the zero address');
```

Comment:
why were we prohibiting this?

the only scenario I can think of where `caller == 0` is in `__execute__`. i guess it would be weird to use this guard in that context considering that the caller is always known (it's `0`)

Reply:
I believe the concern was for scenarios where if an owner renounces (setting owner to `0`), another actor couldn't call protected functions as `0`

Reply:
you meant "could" as in not having the check would open that door, right? but that would only be possible if `assert_only_owner` was being called in `__execute__` for which they should have been able to bypass `__validate__` first.

the other possibility is that `__validate__` uses `assert_only_owner` but again, the developer should be aware by then that the caller will ALWAYS be zero in that context. i'm not sure this is an actual risk.

Reply:
Good points. I'll remove the check

Reply:
just out of extra caution would like to summon @Amxx and @frangio, considering our "make it extremely hard to misuse" design principle

Reply:
This doesn't seem to be about "hard to misuse", but about "defense in depth". Without much context I would keep this check.